### PR TITLE
fix problems introduced in text search

### PR DESCRIPTION
**Pull request by aklt**
* fixes issues introduced by
  https://github.com/devtools-html/debugger.html/pull/3636

Associated Issue: #3638

Note: There is a flow issue with this PR.  Running `yarn flow` gives this error:

```sh
yarn flow v0.24.6
$ flow
Error: src/components/PrimaryPanes/SourcesTree.js:250
                      v------------
250:     const tree = ManagedTree({
251:       key: isEmpty ? "empty" : "full",
252:       getParent: item => {
...:
270:     });
         -^ function call
 38:   props: Props;
              ^^^^^ property `disabledFocus`. Property not found in. See: src/components/shared/ManagedTree.js:38
                                  v
250:     const tree = ManagedTree({
251:       key: isEmpty ? "empty" : "full",
252:       getParent: item => {
...:
270:     });
         ^ object literal


Found 1 error
error Command failed with exit code 2.
```
Which i cannot find a good solution to and I need some sleep. :-)

### CODE REVIEW

File: src/components/ProjectSearch/TextSearch.js [hunk 1]
```diff
@@ -153,28 +160,23 @@ export default class TextSearch extends Component {
       }
     });
 
-    return (
-      <span className="line-value">
-        {matches}
-      </span>
-    );
+    return dom.span({ className: "line-value" }, ...matches);
```
Comment (bomsy) [inline src/components/ProjectSearch/TextSearch.js]: Why did you need to change this back?

File: src/components/shared/ManagedTree.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ type Props = {
   onExpand?: (item: any) => void,
   onCollapse?: (item: any) => void,
   renderItem: any,
-  disabledFocus: boolean
+  disabledFocus?: boolean
```
Comment (jasonLaster) [inline src/components/shared/ManagedTree.js]: i don't know why this is needed now... but it looks good

File: src/components/ProjectSearch/TextSearch.js (hunk 1 above)
Comment (jasonLaster) [inline src/components/ProjectSearch/TextSearch.js]: You get this error when it's an array

```js
    return (
      <span className="line-value">
        {matches}
      </span>
    );
```
screen shot 2017-08-12 at 9 45 03 am

Comment (aklt) [inline src/components/ProjectSearch/TextSearch.js]: Good question.  There is an issue where `<span>{matches}</span>` causes a react
error that keys should be unique,.. oh! I see jasonLaster just commented about that :)

File: src/components/shared/ManagedTree.js (hunk 2 above)
Comment (aklt) [inline src/components/shared/ManagedTree.js]: This was an attempt at making the prop optional to try and make the error mentioned in the PR
above go away.  The `disabledFocus` prop is not used in `src/components/PrimaryPanes/SourcesTree.js` so it should be optional for now.

Unfortunately making it optional did not make `yarn flow` happy.  I think this might be a known flow issue.

Comment (jasonLaster) [inline src/components/shared/ManagedTree.js]: my flow is happy. and so is CIs so you're off the hook

File: src/components/ProjectSearch/TextSearch.js (hunk 1 above)
Comment (jasonLaster) [inline src/components/ProjectSearch/TextSearch.js]: I think we should be able to spread, but it doesn't work for me...

https://github.com/Microsoft/TypeScript/pull/11591/files#diff-36d63f88d66eddff8e2d469bea89b21cR23

Comment (jasonLaster) [inline src/components/ProjectSearch/TextSearch.js]: looks like there are some issues with spreading and it is a performance hit for react as well.
https://github.com/babel/babylon/pull/42

With that in mind, lets merge this and then refactor matches properly

Here's a gist of a patch i started https://gist.github.com/jasonLaster/11e46330c6cc17b36c4f78cb3229fa30

File: src/components/shared/ManagedTree.js (hunk 2 above)
Comment (aklt) [inline src/components/shared/ManagedTree.js]: Gah!  So there is something strange in my setup, grrr.

File: src/components/ProjectSearch/TextSearch.js (hunk 1 above)
Comment (aklt) [inline src/components/ProjectSearch/TextSearch.js]: Ah, interesting, so using spreads for children is off the table for now I guess.

Thanks for the links
